src/text-editor-component.coffee

```diff
@@ -119,6 +119,14 @@ class TextEditorComponent
     @gutterContainerComponent?.destroy()
     @domElementPool.clear()
 
+    @verticalScrollbarComponent.destroy()
+    @horizontalScrollbarComponent.destroy()
+
+    @verticalScrollbarComponent = null
+    @horizontalScrollbarComponent = null
+    @onVerticalScroll = null
+    @onHorizontalScroll = null
```

Comment:
These references needs to be cleared out in order to allow `TextEditorComponent` to be freed up. Not sure what's causing them to retain `TextEditorComponent` as, although there's a cycle between references in there, I'd expect V8 mark-and-sweep GC to handle this properly. Maybe a problem with Coffeescript itself or a bug in V8 GC?